interpreter/cling/lib/Utils/ASTImportSource.cpp

```diff
@@ -0,0 +1,189 @@
+        std::vector < NamedDecl * > declVector;
+        declVector.push_back((NamedDecl *) importedDecl);
+        llvm::ArrayRef < NamedDecl * > FoundDecls(declVector);
+        SetExternalVisibleDeclsForName(childCurrentDeclContext,
+                                       ((NamedDecl *) importedDecl)->getDeclName(),
+                                       FoundDecls);
+
+        /* And also put the Decl I found from the parent Interpreter
+         * in the map of the child Interpreter to have it for the future. */
+        m_DeclName_map[childDeclName.getAsString()] = parentDeclName;
+      }
+    }
+
+    void ASTImportSource::ImportDeclContext(DeclContext *declContextToImport,
+                                            ASTImporter &importer,
+                                            DeclarationName &childDeclName,
+                                            DeclarationName &parentDeclName,
+                                            const DeclContext *childCurrentDeclContext) {
+
+      DeclContext *importedDeclContext =
+        importer.ImportContext(declContextToImport);
+
+      if (importedDeclContext) {
+        /* And also put the declaration context I found from the parent Interpreter
+         * in the map of the child Interpreter to have it for the future. */
+        m_DeclContexts_map[importedDeclContext] = declContextToImport;
+
+        /* Not used for now.
+         * m_DeclContextsNames_map[childDeclName.getAsString()] =
+         * std::make_pair(importedDeclContext,declContextToImport); */
+        importedDeclContext->setHasExternalVisibleStorage(true);
+
+        std::vector < NamedDecl * > declVector;
+        declVector.push_back((NamedDecl *) importedDeclContext);
+        llvm::ArrayRef < NamedDecl * > FoundDecls(declVector);
+        SetExternalVisibleDeclsForName(childCurrentDeclContext,
+                                       ((NamedDecl *) importedDeclContext)->getDeclName(),
+                                       FoundDecls);
+
+        /* And also put the Decl I found from the parent Interpreter
+         * in the map of the child Interpreter to have it for the future. */
+        m_DeclName_map[childDeclName.getAsString()] = parentDeclName;
+      }
+    }
+
+    bool ASTImportSource::Import(DeclContext::lookup_result lookup_result,
+                                 ASTContext &from_ASTContext,
+                                 ASTContext &to_ASTContext,
+                                 const DeclContext *childCurrentDeclContext,
+                                 DeclarationName &childDeclName,
+                                 DeclarationName &parentDeclName) {
+
+      /* Check if we found this Name in the parent interpreter */
+      if (lookup_result.empty())
+        return false;
+
+      /* Prepare to import the Decl(Context)  we found in the
+       * child interpreter */
+      const FileSystemOptions systemOptions;
+      FileManager fm(systemOptions, nullptr);
```

Comment:
Why do you need to create your own `FileManager`? I'd assume that you should be using either the one of the parent or the child. But then again I don't know why the importer needs a FileManager. Could you add a comment?